.github/workflows/main.yaml

```diff
@@ -9,7 +9,261 @@ on:
       - main
 
 jobs:
+  changes:
+    runs-on: ubuntu-latest
+    outputs:
+      excluded_module_arg: ${{ steps.analyzer.outputs.excluded_module_arg }}
+    steps:
+      - uses: actions/checkout@v4
+      # collect changed file group by module
+      - uses: tj-actions/changed-files@v44
+        id: filter
+        with:
+          # each module contained IT should list the dir and direct/indirect dependent module's dir in the files_yaml
+          # the variable of module name should replace path separator with '__' and '-' should replace with '_'
+          files_yaml: |
```

Comment:
It is not ideal that this list has to be maintained manually, but at least we have to be sure it is up to date. If someone will add new module to the repository, but not to this list, will this build fail (it should)?

Reply:
good question!!

```main.yaml``` will skip all IT now
after this changed,if new module not add to this list,may cause build fail for the first commit and any future commit relative to the new module

if merge the pr now,we will need some one time job for the new module(update this list)

Reply:
by the way,due to the new module added is not usally happened,the work to update this list may be acceptable
WDYT?